flow360/plugins/report/report_items.py

```diff
@@ -1614,6 +1673,45 @@ def _get_legend(self, context: ReportContext):
             return legend_filename
         return None
 
+    def _check_3d_caption_validity(self, cases):
+        """Checks the validity of caption input for a given Chart3D instance."""
+
+        if isinstance(self.caption, PatternCaption):
+            if self.items_in_row is not None:
+                raise ValueError(
+                    "PatternCaption is not supported Chart3D when items_in_row is not None."
+                )
+
+        if isinstance(self.caption, List):
+            if self.items_in_row is not None:
+                raise ValueError(
+                    "List of captions is not supported for Chart3D when items_in_row is not None."
+                )
+            if len(self.caption) != len(cases):
+                raise ValueError("Caption list is not the same length as the list of cases.")
```

Comment:
the check of len(self.caption) can be moved to the parent class then call `super()._check_caption_validity(...)`.